src/include/sof/audio/component_ext.h

```diff
@@ -147,13 +148,46 @@ static inline int comp_trigger_remote(struct comp_dev *dev, int cmd)
 	return idc_send_msg(&msg, IDC_BLOCKING);
 }
 
+static inline int comp_trigger_local(struct comp_dev *dev, int cmd)
+{
+	int ret;
+
+	ret = dev->drv->ops.trigger(dev, cmd);
+
+	/* start a thread in case of shared component or DP scheduling */
+	if (dev->task) {
+		/* schedule or cancel task, component must be shared or in DP domain */
+		assert((dev->is_shared) ||
+		       (dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP));
```

Comment:
both internal parentheses are superfluous and at least the former ones really cannot even cause any confusion